### PR TITLE
adds method to read test metrics

### PR DESCRIPTION
**Pull request by sam-io**

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -101,14 +102,14 @@ func NewClientAPI(apiURL string, accessToken string) ClientAPI {
 
 func (client *Client) createResource(
 	resource interface{}, resourceType string, resourceName string, endpoint string) (*response, error) {
-	DebugF(1, "creating %s %s", resourceType, resourceName)
```
Comment (ewilde) [inline client.go]: Need to call DebugF

File: test.go [hunk 2]
```diff
@@ -154,6 +184,53 @@ func (client *Client) DeleteTest(test *Test) error {
...
+		region = "all"
+	}
+	if timeframe == "" {
+		timeframe = "month"
+	}
+	if environmentUUID == "" {
+		environmentUUID = "all"
+	}
+
+	log.Printf("[DEBUG] reading %s %s", "metrics", test.ID)
```
Comment (ewilde) [inline test.go]: DebugF

File: test.go [hunk 3]
```diff
@@ -154,6 +183,53 @@ func (client *Client) DeleteTest(test *Test) error {
...
+func (client *Client) ReadTestMetrics(test *Test, input *ReadMetricsInput) (*TestMetric, error) {
+
+	region := input.Region
+	timeframe := input.Timeframe
+	environmentUUID := input.EnvironemntUUID
+	if region == "" {
+		region = "all"
+	}
+	if timeframe == "" {
+		timeframe = "month"
```
Comment (ewilde) [inline test.go]: @sam-io out of interest why do we default to month, api default is day?

Comment (sam-io) [reply test.go]: probably because that is the value that I needed for the project I was working on, the default should be day though

**Pull request merged**